end_to_end/tpu/mixtral/8x7b/2_test_mixtral.sh

```diff
@@ -34,6 +34,9 @@ python3 MaxText/tests/forward_pass_logit_checker.py  MaxText/configs/base.yml lo
 # Test whether the forward pass logits match the golden logits - matmul implementation
 python3 MaxText/tests/forward_pass_logit_checker.py  MaxText/configs/base.yml load_parameters_path=${MATMUL_SCANNED_CHECKPOINT} run_name=matmul_forward_pass_test per_device_batch_size=1 model_name=mixtral-8x7b tokenizer_path=gs://maxtext-external/mixtral-8x7B-v0.1-Instruct/tokenizer.mistral ici_tensor_parallelism=4 ici_fsdp_parallelism=16 max_prefill_predict_length=11 max_target_length=11 dataset_type=synthetic dtype=float32 --atol=3 --rtol=1 --token_size=4
 
+# Test whether the forward pass logits match the golden logits - megablox implementation
+python3 MaxText/tests/forward_pass_logit_checker.py MaxText/configs/base.yml base_output_directory=${BASE_OUTPUT_PATH} load_parameters_path=${MATMUL_SCANNED_CHECKPOINT} run_name=megablox_forward_pass_test per_device_batch_size=1 model_name=mixtral-8x7b tokenizer_path=gs://maxtext-external/mixtral-8x7B-v0.1-Instruct/tokenizer.mistral ici_tensor_parallelism=4 ici_fsdp_parallelism=16 max_prefill_predict_length=11 max_target_length=11 dataset_type=synthetic dtype=float32 moe_matmul=True megablox=True --atol=3 --rtol=1 --token_size=4
```

Comment:
The atol and rtol are very high here. I missed this in the prior PR. Both the atol and rtol should be about 0.1, are the tests not passing for values near 0.1?

Reply:
Based on previous PR, the test was not passing near 0.1. I ran some [exp](https://docs.google.com/spreadsheets/d/1o2RhkRpPXL5WD7FYb9kgm64qjuifOt0E0nfe3dc0YZE/edit?resourcekey=0-HGJ3wSqnzfMQ52O_Pf10sQ&gid=1958907274#gid=1958907274) before, and when number of layers increases, the diverge is huge.

Reply:
Hmm that is quite concerning. Have you tried to run this on a CPU? I think @khatwanimohit could help you try this to see if it is a numeric issue. I think these large atol and rtol allows many wrong implementations to pass this test, which means this isn't a very useful test

Reply:
Yes, I tried on CPU in that sheet. Discussing this offline.